features/mbedtls/targets/TARGET_NUVOTON/TARGET_NUC472/sha/sha_alt_hw.c

```diff
@@ -241,16 +236,16 @@ void crypto_sha_update_nobuf(crypto_sha_context *ctx, const unsigned char *input
     // 1. Last block which may be incomplete
     // 2. Non-last block which is complete
     MBED_ASSERT(islast || ilen == ctx->blocksize);
-    
+
     const unsigned char *in_pos = input;
     int rmn = ilen;
     uint32_t sha_ctl_start = (CRPT->SHA_CTL & ~(CRPT_SHA_CTL_DMALAST_Msk | CRPT_SHA_CTL_DMAEN_Msk)) | CRPT_SHA_CTL_START_Msk;
     uint32_t sha_opmode = (CRPT->SHA_CTL & CRPT_SHA_CTL_OPMODE_Msk) >> CRPT_SHA_CTL_OPMODE_Pos;
-    uint32_t DGST0_old, DGST1_old, DGST2_old, DGST3_old, DGST4_old, DGST5_old, DGST6_old, DGST7_old;
-    
+    uint32_t DGSTs[8] = { 0 };
+
     while (rmn > 0) {
         CRPT->SHA_CTL = sha_ctl_start;
```

Comment:
@ccli8 What's the reason this file doesn't use the driver code from `nuc472_crypto.c`?

Reply:
@hanno-arm The driver code mainly supports DMA mode. Current SHA accelerator implementation supports non-DMA mode to spare DMA buffer. This needs direct register control.